### ISSUE TITLE
[ISSUE] user external_id (optional) erase by null - Azure

### ISSUE THREAD

**Issue by nicolasroy-bib** [databrickslabs/terraform-provider-databricks#1073]
Hello everyone

### Configuration
```hcl
# Import databricks plugin (with min version)
terraform {
  required_providers {
    databricks = {
      source  = "databrickslabs/databricks"
      version = ">= 0.4.5"
    }
  }
}

# Referencing special admins group
data "databricks_group" "admins" {
  display_name     = "admins"
}

# Create user
resource "databricks_user" "new_user" {
  user_name             = var.user_email
  display_name          = var.user_display_name
  allow_cluster_create  = true
}

# Assign user to group
resource "databricks_group_member" "user_become_admin" {
  group_id  = data.databricks_group.admins.id
  member_id = databricks_user.new_user.id
}
```

### Expected Behavior
Ignore external id optional value for user

### Actual Behavior
For every account, an Azure external ID has been found
It reset by null each id during apply. Azure re-set them after automatically

```hcl
   # module.admin_user["xxxxxxxxxxx"].databricks_user.new_user will be updated in-place
  ~ resource "databricks_user" "new_user" {
      - external_id                = "3430ce80-d8b0-4a93-821e-a776b462a65f" -> null
        id                         = "8480986995235910"
        # (7 unchanged attributes hidden)
    }
```

I have try to create a new account, all is ok during the creation
```hcl
  # module.admin_user[xxxxxxxxxx"].databricks_group_member.user_become_admin will be created
  + resource "databricks_group_member" "user_become_admin" {
      + group_id  = "5323687635800966"
      + id        = (known after apply)
      + member_id = (known after apply)
    }
  # module.admin_user["xxxxxxxxxx"].databricks_user.new_user will be created
  + resource "databricks_user" "new_user" {
      + active                     = true
      + allow_cluster_create       = true
      + allow_instance_pool_create = false
      + databricks_sql_access      = false
      + display_name               = "xxxxxxxxxx"
      + id                         = (known after apply)
      + user_name                  = "xxxxxxxxxx
      + workspace_access           = false
    }
```
but when I relaunch a plan, it detects the external_id and indicate to put to null

### Steps to Reproduce
1. `terraform plan`
2. `terraform apply`

### Terraform and provider versions
Terraform v1.0.10
databrickslabs/databricks v0.4.5

### Important Factoids
I have update databrickslabs/databricks from 0.3.5 to 0.4.5
I see that 0.4 add the value external_id but value is set by azure

**Comment by alexott** [comment]
Do you have any additional sync method, like, SCIM Connector?

**Comment by nfx** [comment]
@nicolasroy-bib i could not reproduce this issue on an Azure Databricks workspace and i didn't see Azure setting `external_id` for users created with TF.

you have three options:

1) `lifecycle { ignore_changes = [external_id] }`
2) `databricks_user` data resource to get user id by user name - https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs/data-sources/user
3) `force` attribute to implicitly import the user into workspace, regardless of any SCIM provisioning processes you have setup: https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs/resources/user#argument-reference

**Comment by nicolasroy-bib** [comment]
hum, I think it's due to the link between my company and azure for authentication but not sure.
The first solution seems good, I have test it's ok, 
thanks a lot

**Comment by nfx** [comment]
@nicolasroy-bib it was fixed in #1102 and released in [v0.4.9](https://github.com/databrickslabs/terraform-provider-databricks/releases/tag/v0.4.9)

**Comment by nicolasroy-bib** [comment]
Update done with 0.4.9, and I have suppress the ignore_changes option. all is good now, thanks a lot
